fastestimator/estimator/trace.py

```diff
@@ -911,3 +931,125 @@ def get_logdir(self):
+        else:
+            raise ValueError("Multiple losses defined in Network, specify one loss in monitor_name for save_best_only")
+
+    def _resolve_auto_save_model(self, state):
+        if self.monitor_name in self.network.all_losses:
+            self.save_mode = 'min'
+        else:
+            try:
+                self.save_mode = state['{}_save_mode'.format(self.monitor_name)]
+            except KeyError:
+                raise ValueError("save_mode='auto' cannot be resolved.")
+
+        if self.save_mode == 'min':
+            self.monitor_op = np.less
+            self.best = np.Inf
+        elif self.save_mode == 'max':
+            self.monitor_op = np.greater
+            self.best = -np.Inf
+        else:
+            raise ValueError("save_mode='auto' cannot be resolved.")
+
+        print("FastEstimator-ModelCheckpoint: Save models when '{}' is {}.".format(self.monitor_name, self.save_mode))
+
+    def on_begin(self, state):
+        # TODO: load latest saved model
+        if self.monitor_name is None and self.save_best_only:
+            self._resolve_monitor_name()
+
+        if self.save_mode == 'auto' and self.save_best_only:
+            self._resolve_auto_save_model(state)
+
+        if self.model_names is None:
+            self.model_names = self.network.model.keys()
+        elif isinstance(self.model_names, str):
+            self.model_names = [self.model_names]
+
+        if self.mode == 'train' and self.save_best_only and self.monitor_name in self.network.all_losses:
+            raise ValueError("`mode=train` and `save_best_only=True` only support non-loss metrics.")
+
+    def on_epoch_end(self, state):
+        if state['mode'] == self.mode and (state['epoch'] % self.save_freq == 0):
+            self._save_model(state)
+
+    def _is_best_model(self, state):
+        if self.monitor_op(state[self.monitor_name], self.best):
+            self.best = state[self.monitor_name]
+            return True
+
+        return False
+
+    def _save_model(self, state):
+        if self.save_best_only:
+            is_best_model = self._is_best_model(state)
+
+        # remove existing saved files when save_best_only=True
+        if self._last_saved_files and self.save_best_only and is_best_model:
+            for saved_file in self._last_saved_files:
+                os.remove(saved_file)
+            self._last_saved_files = []
+
+        # save all specified models
+        if (self.save_best_only and is_best_model) or (not self.save_best_only):
+            for model_key in self.model_names:
+                save_path = os.path.join(
+                    self.save_dir, '{}_epoch_{}_step_{}.h5'.format(model_key, state['epoch'], state['train_step']))
+                self.network.model[model_key].save(save_path, include_optimizer=False)
+                self._last_saved_files.append(save_path)
+                print("FastEstimator-ModelCheckpoint: Saving '{}' to {}".format(model_key, save_path))
```

Comment:
Can you add a verbose so that users have a choice to disable printing the message?